dlc/src/main/scala/org/bitcoins/dlc/BinaryOutcomeDLCWithSelf.scala

```diff
@@ -500,6 +490,75 @@ case class BinaryOutcomeDLCWithSelf(
     }
   }
 
+  def executeJusticeDLC(
+      dlcSetup: SetupDLC,
+      timedOutCET: Transaction,
+      local: Boolean): Future[DLCOutcome] = {
+    val justiceOutput = timedOutCET.outputs.head
+    val normalOutput = timedOutCET.outputs.last
+
+    val cetPrivKey = if (local) {
+      if (timedOutCET == dlcSetup.cetWinRemote) {
+        cetLocalWinPrivKey
+      } else {
+        cetLocalLosePrivKey
+      }
+    } else {
+      if (timedOutCET == dlcSetup.cetWinLocal) {
+        cetRemoteWinPrivKey
+      } else {
+        cetRemoteLosePrivKey
+      }
+    }
+
+    val justiceSpendingInfo = ConditionalSpendingInfo(
+      TransactionOutPoint(timedOutCET.txIdBE, UInt32.zero),
+      justiceOutput.value,
+      justiceOutput.scriptPubKey.asInstanceOf[ConditionalScriptPubKey],
+      Vector(cetPrivKey),
+      HashType.sigHashAll,
+      ConditionalPath.nonNestedFalse
+    )
+
+    val normalSpendingInfo = P2PKHSpendingInfo(
+      TransactionOutPoint(timedOutCET.txIdBE, UInt32.one),
+      normalOutput.value,
+      normalOutput.scriptPubKey.asInstanceOf[P2PKHScriptPubKey],
+      cetPrivKey,
+      HashType.sigHashAll
+    )
+
+    val finalPrivKey = if (local) {
+      finalLocalPrivKey
+    } else {
+      finalRemotePrivKey
+    }
+
+    val justiceSpendingTxF =
+      constructClosingTx(justiceOutput,
```

Comment:
isn't it a bit redundant to pass in `justiceOutput` here when it is already inside of `justiceSpendingInfo`? It looks like `cetOuptut` isn't even really used in `constructClosingTx`